src/renderer/helpers/api/local.js

```diff
@@ -1009,6 +1075,48 @@ export function parseLocalListVideo(item) {
       liveNow: false,
       isUpcoming: false,
     }
+  } else if (item.type === 'GridVideo') {
+    /** @type {import('youtubei.js').YTNodes.GridVideo} */
+    const video = item
+
+    let publishedText
+
+    if (video.published != null && !video.published.isEmpty()) {
+      publishedText = video.published.text
+    }
+
+    const published = calculatePublishedDate(
+      publishedText,
+      video.is_live,
+      video.is_upcoming || video.is_premiere,
+      video.upcoming
+    )
+
+    return {
+      type: 'video',
+      videoId: video.id,
+      title: video.title.text,
+      author: video.author?.name,
+      authorId: video.author?.id,
+      viewCount: video.views.text == null ? null : extractNumberFromString(video.views.text),
+      published,
+      lengthSeconds: Number(timestampToDuration(video.duration.text)),
```

Comment:
As this is just for the local API you could use YouTube.js' `Utils.timeToSeconds()` function, instead of introducing your own one.